File: backend/app/models/news.py
from datetime import datetime
from enum import Enum
from typing import Optional
from sqlmodel import Field, SQLModel

class NewsCategory(Enum):
    RUMOR='Rumor'
    SENS= 'Sens'
    ARTICLE='Article'
    RULING='Ruling'


class News(SQLModel,table=True):
    id:Optional[int] = Field(default=None,primary_key=True)
    timestamp:datetime=Field(nullable=False)
    category:NewsCategory=Field(nullable=False)
    source:Optional[str] = Field(nullable=False)
    author:Optional[str] =Field(nullable=False)
    full_story:str =Field(nullable=False)
    ticker:str=Field(nullable=False)

File: frontend/lib/types/news.ts
import { z } from 'zod';

const NewsCategoryEnum = z.enum(['Rumor', 'Sens', 'Article', 'Ruling']);

const NewsItemSchema = z.object({
  id: z.number().int(),
  timestamp: z.coerce.date(),
  category: NewsCategoryEnum,
  description: z.string(),
  source: z.string().nullable().optional(),
  author: z.string().nullable().optional(),
  full_story: z.string(),
});

export const NewsResponseSchema = z.object({
  news_items: z.array(NewsItemSchema),
});

export type NewsItem = z.infer<typeof NewsItemSchema>;
export type NewsResponse = z.infer<typeof NewsResponseSchema>;